src/device.rs

```diff
@@ -0,0 +1,78 @@
+use bit_field::BitArray;
+use embedded_hal::digital::v2::OutputPin;
+
+use crate::bus::{Bus, FourWire, ThreeWire};
+use crate::host::Host;
+use crate::register;
+use crate::socket::Socket;
+use crate::uninitialized_device::UninitializedDevice;
+
+pub struct Device<SpiBus: Bus, HostImpl: Host> {
```

Comment:
Shouldnt this be called something like `W5500` instead of generically calling it `Device`?

Reply:
I felt that was kind of repetitive, considering the crate is already called `w5500`.  `w5500::W5500`.  IMO, I'm willing to discuss renaming.